app/controllers/v0/sessions_controller.rb

```diff
@@ -16,22 +16,14 @@ class SessionsController < ApplicationController
     # @type is set automatically by the routes in config/routes.rb
     # For more details see SAML::SettingsService and SAML::URLService
     def new
-      type = params[:signup] ? 'signup' : params[:type]
-      if REDIRECT_URLS.include?(type)
-        url = url_service.send("#{type}_url")
-
-        # If a clientId param exists, include GA clientId for cross-domain analytics
-        client_id = params[:clientId]
-        url = client_id ? "#{url}&clientId=#{client_id}" : url
-
-        if type == 'slo'
-          Rails.logger.info('SSO: LOGOUT', sso_logging_info)
-          reset_session
-        end
-        redirect_to url
-      else
-        raise Common::Exceptions::RoutingError, params[:path]
+      type = params[:type]
+      raise Common::Exceptions::RoutingError, params[:path] unless REDIRECT_URLS.include?(type)
+      url = url_service(new_query_params).send("#{type == 'signup' ? 'idme' : type}_url")
```

Comment:
Although I was hoping to get rid of `signup_url` from `URLService`, if we're going to implement a distinct `signup` route, we'll either need this conditional or keep the `signup_url`. I was thinking that if it's cleaner to keep `signup_url`, perhaps we could `raise Common::Exceptions::Forbidden` or something if the `URLService` instance invoking that method doesn't have `op: signup` in the query.